### PR TITLE
Feature/localization

### LINKED ISSUE #135: Localization
At the moment, the app is only available in English. To reach our target group here in Germany (= non-tech savvy users), we'd actually need a German version as well. Ideally, the language would be chosen based on the operating system's language: German, if the phone is set to German and English in all other cases.

- [ ] quick research: What (simple) libraries are there? Does Ionic already provide some localization mechanisms?
- [ ] implement localization to German

### PR DESCRIPTION
**Pull request by schrooom**
Closes #135

### CODE REVIEW

File: src/app/shared-services/inferences/engine/types.ts [hunk 1]
```diff
@@ -20,15 +21,18 @@ export class InferenceDefinition {
...
+    public info: (res: Inference, translate: TranslateService) => string,
     public scoringConfigurations: InferenceScoringConfig[]
   ) {}
 
   public getScoringConfig(type: InferenceScoringType): InferenceScoringConfig {
     return this.scoringConfigurations.find((config) => config.type === type)
   }
 
+  public getName(translateSerice: TranslateService): string {
+    return translateSerice.instant(this.type)
```
Comment (jbraese) [inline src/app/shared-services/inferences/engine/types.ts]: there is a v missing in the variable name

File: src/assets/i18n/de.json [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+      "createInferencesButtonText": "Rückschlüsse generieren"
+    }
+  },
+  "tracking": {
+    "title": "Standortverlauf",
+    "toggleTitle": "Aufzeichnung starten / stoppen",
+    "stateRunning": "Läuft",
+    "stateStopped": "Gestoppt",
+    "notSupported": "Auf dieser Plattform wird die Aufzeichnung des Standortes nicht unterstützt",
+    "androidBatteryHint": "Damit die App korrekt funktioniert solltest du in den Einstellungen die Batterieoptimierung für diese App deaktivieren. Weitere Informationen findest du unter <a href='https://dontkillmyapp.com'>dieser Seite</a>",
```
Comment (jbraese) [inline src/assets/i18n/de.json]: there is a comma missing after "funktioniert". I would add a full stop at the end here. I would also change "findest du unter dieser Seite" to "findest du auf dieser Seite".

**Review by jbraese — APPROVED** [review]
Looks great! It must have been a lot of work to find all these things. 

I found a few still untranslated things in the app (all are English being displayed in the German version):

Map page
- the attribution should change from "OSM contributors" to "OpenStreetMap-Mitwirkende"
- when clicking on the icons for home/work in the map, the English inference type is still displayed, eg "home (mittlere Wahrscheinlichkeit)". 
- when clicking on the last collected point, the text displayed is still in English, eg "Timestamp: 10/2/2021, 7:06-36 PM"

Explore page
- "Erkundung" is a bit clunky in my opinion and isn't really specific to the page's content. Then again, neither is "Explore". What do you think about just changing the whole page name to "Timetable"/"Zeitplan"?

Diary page
- the date does not seem to be internationalized (English abbreviations for the months).
- when creating a new entry and manually choosing the date, the buttons to finish date selection are still in English ("cancel", "done").

**Comment by schrooom** [comment]
Thank you @jbraese for your thorough review, much appreciated! The things mentioned by you are mostly done with my two latest commits.

Nevertheless I spotted another issue regarding the date-time-localization: The translation is not always falling back to our default language (english), but is using the actual device-language (e.g. dutch). I'll try to sort that out by tomorrow :) **→ Done!**

**Comment by zven** [comment]
@schrooom Apart from the failing test this feature would be ready to be merged into `develop`, right?

**Comment by schrooom** [comment]
> @schrooom Apart from the failing test this feature would be ready to be merged into `develop`, right?

Correct.

**Pull request merged**